### ISSUE TITLE
Importing large macro sets can take an exceedingly long time.

### ISSUE THREAD

**Issue by Phergus** [RPTools/maptool#1483]
**Describe the bug**
Large macro sets can take an exceedingly long time to import and during the import there is no indication within the GUI that anything is happening.

**To Reproduce**
1. Create 100 macros on a token.
2. Export macros as a macro set.
3. On new token, import exported 100 macro set.
4. Can take several seconds to import with no feedback.

**Expected behavior**
Change mouse pointer to wait symbol or other visual indication that an action is taking place.  Improve import speed if possible.

**MapTool Info**
- Version: Any
- Install: Any

**Desktop:**
 - OS: Windows
 - Version: 10

**Additional context**
Macro to create 100 macros on a token.  Use Apply to Selected.
```
[h: colors = '["red", "green", "blue", "white", "yellow", "orange", "cyan", "gray", "lime", "magenta", "teal"]']
[h, count(100,""), code:{
	[h: createMacro("Button "+ roll.count, "My name is: [r: token.name]", "color="+json.get(colors,1d11-1))]
}]
```

**Comment by dorpond** [comment]
Here is a file that you can use to import. This sucker took me so long that I thought MT crashed (and because there was nothing that said "Importing file".
[New Macro Set 4-7-2020.zip](https://github.com/RPTools/maptool/files/4451752/New.Macro.Set.4-7-2020.zip)

**Comment by Phergus** [comment]
Fixed under #2118.
